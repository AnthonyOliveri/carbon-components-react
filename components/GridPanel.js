import React from 'react';
import PropTypes from 'prop-types';
import Grid from './Grid';
import GridPanelSettings from './GridPanelSettings';
import Pagination from './Pagination';
import Icon from './Icon';

export default class GridPanel extends React.Component {
  static copyColumns(columns) {
    return columns.map(column => Object.assign({}, column));
  }

  state = {
    columns: [],
  };

  componentWillMount() {
    this.initiateColumns(this.props.columns);
  }

  componentWillReceiveProps(nextProps) {
    this.updateColumns(nextProps.columns, 'render');
  }

  getColumns() {
    const shownColumns = [];
    const hideableColumns = [];

    this.state.columns.forEach(column => {
      if (column.hideable) {
        hideableColumns.push(column);
      }
      if (!column.hidden) {
        shownColumns.push(column);
      }
    });

    return {
      hideableColumns,
      shownColumns,
    };
  }

  initiateColumns(columns) {
    let stateColumns = GridPanel.copyColumns(columns);

    if (this.props.localStorageKey && this.props.name) {
      const config = JSON.parse(
        localStorage.getItem(this.props.localStorageKey) || '{}'
      ); // eslint-disable-line no-undef
      if (
        config &&
        config[this.props.name] &&
        config[this.props.name].columns
      ) {
        stateColumns = columns.map((column, index) =>
          Object.assign({}, column, config[this.props.name].columns[index])
        );
      }
    }

    this.setState(
      {
        columns: stateColumns,
      },
      this.saveTableConfig
    );
  }

  resetColumns(updatedColumns) {
    this.setState(
      {
        columns: GridPanel.copyColumns(updatedColumns),
      },
      this.saveTableConfig
    );
    this.props.resetGrid();
  }

  saveTableConfig() {
    if (this.props.localStorageKey) {
      const config = JSON.parse(
        localStorage.getItem(this.props.localStorageKey) || '{}'
      ); // eslint-disable-line no-undef
      config[this.props.name] = Object.assign({}, config[this.props.name], {
        columns: this.state.columns,
      });
      localStorage.setItem(this.props.localStorageKey, JSON.stringify(config)); // eslint-disable-line no-undef
    }
  }

  toggleColumnVisibility(updatedColumn) {
    const columns = this.state.columns.map(column => {
      if (column.name === updatedColumn.name) {
        return Object.assign({}, column, {
          hidden: !column.hidden,
        });
      }
      return column;
    });
    this.setState(
      {
        columns: GridPanel.copyColumns(columns),
      },
      this.saveTableConfig
    );
  }

  updateColumns(updatedColumns, key) {
    const updatedColumnsMap = new Map();
    updatedColumns.forEach(column => {
      updatedColumnsMap.set(column.name, column);
    });
    const columns = this.state.columns.map(column => {
      const updatedColumn = updatedColumnsMap.get(column.name);
      if (updatedColumn) {
        return Object.assign({}, column, {
          [key]: updatedColumn[key],
        });
      }
      return column;
    });
    this.setState(
      {
        columns: GridPanel.copyColumns(columns),
      },
      this.saveTableConfig
    );
  }

  render() {
    let pagination = '';
    const { hideableColumns, shownColumns } = this.getColumns();

    if (this.props.showPager) {
      pagination = (
        <div className="bx--grid-panel__pagination">
          <Pagination
            backwardText={this.props.backwardText}
            className={this.props.paginationClassName}
            itemRangeText={this.props.itemRangeText}
            forwardText={this.props.forwardText}
            itemsPerPageText={this.props.itemsPerPageText}
            onChange={this.props.changePage}
            page={this.props.page}
            pageNumberText={this.props.pageNumberText}
            pageRangeText={this.props.pageRangeText}
            pageSizes={this.props.pageSizes}
            totalItems={this.props.totalItems}
          />
          <button
            className="bx--grid-panel__refresh"
            onClick={this.props.refreshGrid}
          >
            <Icon name="restart--glyph" description="Refresh" />
          </button>
          <GridPanelSettings
            items={hideableColumns}
            resetColumns={() => this.resetColumns(this.props.columns)}
            toggleColumnVisibility={column =>
              this.toggleColumnVisibility(column)}
          />
        </div>
      );
    }

    return (
      <div className="bx--grid-panel">
        {pagination}
        <Grid
          columns={shownColumns}
          data={this.props.data}
          detailPanelWidth={this.props.detailPanelWidth}
          emptyText={this.props.emptyText}
          expandedDetailIndex={this.props.expandedDetailIndex}
          expandedRowIndices={this.props.expandedRowIndices}
          isFetching={this.props.isFetching}
          onSort={this.props.onSort}
          renderDetailExpansion={this.props.renderDetailExpansion}
          renderRowExpansion={this.props.renderRowExpansion}
          sort={this.props.sort}
          updateColumns={updatedColumns =>
            this.updateColumns(updatedColumns, 'width')}
        />
      </div>
    );
  }
}

GridPanel.defaultProps = {
  changePage: () => {},
  expandedDetailIndex: null,
  expandedRowIndices: [],
  isFetching: false,
  loadNextPage: () => {},
  loadPreviousPage: () => {},
  name: '',
  onSort: () => {},
  pageSizes: [25, 50, 75, 100, 200],
  columns: [],
  data: [],
  paginationClassName: '',
  renderDetailExpansion: () => {},
  renderRowExpansion: () => {},
  refreshGrid: () => {},
  resetGrid: () => {},
  showPager: true,
  sort: {},
  totalItems: 0,
};

GridPanel.propTypes = {
  backwardText: PropTypes.string,
  changePage: PropTypes.func,
  columns: PropTypes.arrayOf(
    PropTypes.shape({
      hideable: PropTypes.bool,
      hidden: PropTypes.bool,
      name: PropTypes.string.isRequired,
      overflow: PropTypes.bool,
      render: PropTypes.func.isRequired,
      sortable: PropTypes.bool,
      title: PropTypes.string,
      width: PropTypes.number,
    })
  ).isRequired,
  data: PropTypes.array,
  detailPanelWidth: PropTypes.number,
  emptyText: PropTypes.string,
  expandedDetailIndex: PropTypes.number,
  expandedRowIndices: PropTypes.arrayOf(PropTypes.number),
  forwardText: PropTypes.string,
  isFetching: PropTypes.bool,
  itemRangeText: PropTypes.func,
  itemsPerPageText: PropTypes.string,
  localStorageKey: PropTypes.string,
  name: PropTypes.string.isRequired,
  onSort: PropTypes.func,
  page: PropTypes.number.isRequired,
  pageNumberText: PropTypes.string,
  pageRangeText: PropTypes.func,
  pageSizes: PropTypes.arrayOf(PropTypes.number),
  paginationClassName: PropTypes.string,
  renderDetailExpansion: PropTypes.func,
  renderRowExpansion: PropTypes.func,
  refreshGrid: PropTypes.func,
  resetGrid: PropTypes.func,
  showPager: PropTypes.bool,
  sort: PropTypes.shape({
    direction: PropTypes.string,
    property: PropTypes.string,
  }),
  totalItems: PropTypes.number.isRequired,
};
