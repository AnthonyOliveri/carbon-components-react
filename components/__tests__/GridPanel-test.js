import GridPanel from '../GridPanel';
import React from 'react';
import { shallow, mount } from 'enzyme';

jest.unmock('../GridPanel');

const shallowRender = props => shallow(<GridPanel {...props} />);
const fullRender = props => mount(<GridPanel {...props} />);

describe('GridPanel', () => {
  describe('check if props are rendered correctly', () => {
    const columns = [
      {
        name: 'column1',
        title: 'column1',
        render: rowData => rowData.column1,
        width: 50,
      },
      {
        name: 'column2',
        title: 'column2',
        sortable: true,
        render: rowData => rowData.column2,
        width: 25,
      },
      {
        name: 'column3',
        title: 'column3',
        render: rowData => rowData.column3,
        width: 25,
      },
    ];
    describe('standard grid - loaded', () => {
      const data = [
        {
          column1: 'stuff1',
          column2: 'more1',
          column3: 'test1',
        },
        {
          column1: 'stuff2',
          column2: 'more2',
          column3: 'test2',
        },
        {
          column1: 'stuff3',
          column2: 'more3',
          column3: 'test3',
        },
      ];
      const wrapper = shallowRender({
        changePage: jest.fn(),
        emptyText: 'emptyyy',
        expandedDetailIndex: null,
        expandedRowIndices: [],
        isFetching: false,
        localStorageKey: 'testGrid1',
        name: '',
        onSort: jest.fn(),
        page: 1,
        pageSizes: [5, 10, 15],
        refreshGrid: jest.fn(),
        resetGrid: jest.fn(),
        totalItems: 3,
        columns,
        data,
        renderDetailExpansion: jest.fn(),
        renderRowExpansion: jest.fn(),
        showPager: true,
        sort: {
          direction: 'ASC',
          property: 'column2',
        },
      });

      it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('standard grid - fetching', () => {
      const wrapper = shallowRender({
        changePage: jest.fn(),
        emptyText: 'emptyyy',
        expandedDetailIndex: null,
        expandedRowIndices: [],
        isFetching: true,
        localStorageKey: 'testGrid2',
        name: '',
        onSort: jest.fn(),
        page: 1,
        refreshGrid: jest.fn(),
        resetGrid: jest.fn(),
        totalItems: 3,
        columns,
        data: [],
        renderDetailExpansion: jest.fn(),
        renderRowExpansion: jest.fn(),
        showPager: true,
        sort: {
          direction: 'ASC',
          property: 'column2',
        },
      });

      it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('standard grid - empty', () => {
      const wrapper = shallowRender({
        changePage: jest.fn(),
        emptyText: 'emptyyy griddd',
        expandedDetailIndex: null,
        expandedRowIndices: [],
        isFetching: false,
        localStorageKey: 'testGrid3',
        name: '',
        onSort: jest.fn(),
        page: 1,
        refreshGrid: jest.fn(),
        resetGrid: jest.fn(),
        totalItems: 3,
        totalPages: 1,
        columns,
        data: [],
        renderDetailExpansion: jest.fn(),
        renderRowExpansion: jest.fn(),
        showPager: true,
        sort: {
          direction: 'ASC',
          property: 'column2',
        },
      });

      it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('standard grid - rows expanded', () => {
      const data = [
        {
          column1: 'stuff1',
          column2: 'more1',
          column3: 'test1',
        },
        {
          column1: 'stuff2',
          column2: 'more2',
          column3: 'test2',
          expander: 'expand2',
        },
        {
          column1: 'stuff3',
          column2: 'more3',
          column3: 'test3',
          expander: 'expand3',
        },
      ];
      const wrapper = shallowRender({
        changePage: jest.fn(),
        emptyText: 'emptyyy griddd',
        expandedDetailIndex: null,
        expandedRowIndices: [1, 2],
        isFetching: false,
        localStorageKey: 'testGrid4',
        name: '',
        onSort: jest.fn(),
        page: 1,
        refreshGrid: jest.fn(),
        resetGrid: jest.fn(),
        totalItems: 3,
        columns,
        data,
        renderDetailExpansion: jest.fn(),
        renderRowExpansion: (rowData, index) => `${rowData.expander}-${index}`,
        showPager: true,
        sort: {
          direction: 'ASC',
          property: 'column2',
        },
      });

      it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('standard grid - detail expanded', () => {
      const data = [
        {
          column1: 'stuff1',
          column2: 'more1',
          column3: 'test1',
        },
        {
          column1: 'stuff2',
          column2: 'more2',
          column3: 'test2',
          expander: 'expand2',
        },
        {
          column1: 'stuff3',
          column2: 'more3',
          column3: 'test3',
          expander: 'expand3',
        },
      ];
      const wrapper = shallowRender({
        changePage: jest.fn(),
        emptyText: 'emptyyy griddd',
        expandedDetailIndex: 2,
        expandedRowIndices: [],
        isFetching: false,
        localStorageKey: 'testGrid5',
        name: '',
        onSort: jest.fn(),
        page: 1,
        refreshGrid: jest.fn(),
        resetGrid: jest.fn(),
        totalItems: 3,
        columns,
        data,
        renderDetailExpansion: rowData =>
          `${rowData.column1}-${rowData.column3}-${rowData.column2}-${rowData.expander}`,
        renderRowExpansion: jest.fn(),
        showPager: true,
        sort: {
          direction: 'ASC',
          property: 'column2',
        },
      });

      it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('standard grid - hidden pagination toolbar', () => {
      const data = [
        {
          column1: 'stuff1',
          column2: 'more1',
          column3: 'test1',
        },
        {
          column1: 'stuff2',
          column2: 'more2',
          column3: 'test2',
        },
        {
          column1: 'stuff3',
          column2: 'more3',
          column3: 'test3',
        },
      ];
      const wrapper = shallowRender({
        changePage: jest.fn(),
        emptyText: 'emptyyy',
        expandedDetailIndex: null,
        expandedRowIndices: [],
        isFetching: false,
        name: '',
        onSort: jest.fn(),
        page: 1,
        pageSizes: [5, 10, 15],
        refreshGrid: jest.fn(),
        resetGrid: jest.fn(),
        totalItems: 3,
        columns,
        data,
        renderDetailExpansion: jest.fn(),
        renderRowExpansion: jest.fn(),
        showPager: false,
        sort: {
          direction: 'ASC',
          property: 'column2',
        },
      });

      it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });
  });

  describe('check if the actions are triggered correctly', () => {
    describe('grid gets initialized from localStorage', () => {
      localStorage.setItem(
        'testLocal',
        JSON.stringify({
          // eslint-disable-line no-undef
          testGrid13: {
            columns: [
              {
                name: 'column1',
                title: 'column1',
                width: 35,
              },
              {
                name: 'column2',
                title: 'column2',
                sortable: true,
                width: 40,
              },
              {
                name: 'column3',
                title: 'column3',
                width: 25,
              },
            ],
          },
        })
      );
      const columns = [
        {
          name: 'column1',
          title: 'column1',
          render: rowData => rowData.column1,
          width: 50,
        },
        {
          name: 'column2',
          title: 'column2',
          sortable: true,
          render: rowData => rowData.column2,
          width: 25,
        },
        {
          name: 'column3',
          title: 'column3',
          render: rowData => rowData.column3,
          width: 25,
        },
      ];
      const data = [
        {
          column1: 'stuff1',
          column2: 'more1',
          column3: 'test1',
        },
        {
          column1: 'stuff2',
          column2: 'more2',
          column3: 'test2',
        },
        {
          column1: 'stuff3',
          column2: 'more3',
          column3: 'test3',
        },
      ];
      const wrapper = fullRender({
        changePage: jest.fn(),
        emptyText: 'emptyyy',
        expandedDetailIndex: null,
        expandedRowIndices: [],
        isFetching: false,
        localStorageKey: 'testLocal',
        name: 'testGrid13',
        onSort: jest.fn(),
        page: 1,
        refreshGrid: jest.fn(),
        resetGrid: jest.fn(),
        totalItems: 3,
        columns,
        data,
        renderDetailExpansion: jest.fn(),
        renderRowExpansion: jest.fn(),
        showPager: true,
        sort: {
          direction: 'ASC',
          property: 'column2',
        },
      });

      it('initiateColumns sets the correct state', () => {
        columns[0].width = 35;
        columns[1].width = 40;
        columns[2].width = 25;

        expect(wrapper.state().columns).toEqual(columns);
      });
    });

    describe('props gets updated', () => {
      const columns = [
        {
          name: 'column1',
          title: 'column1',
          render: rowData => rowData.column1,
          width: 50,
        },
        {
          name: 'column2',
          title: 'column2',
          sortable: true,
          render: rowData => rowData.column2,
          width: 25,
        },
        {
          name: 'column3',
          title: 'column3',
          render: rowData => rowData.column3,
          width: 25,
        },
      ];
      const data = [
        {
          column1: 'stuff1',
          column2: 'more1',
          column3: 'test1',
        },
        {
          column1: 'stuff2',
          column2: 'more2',
          column3: 'test2',
        },
        {
          column1: 'stuff3',
          column2: 'more3',
          column3: 'test3',
        },
      ];
      const wrapper = fullRender({
        changePage: jest.fn(),
        emptyText: 'emptyyy',
        expandedDetailIndex: null,
        expandedRowIndices: [],
        isFetching: false,
        localStorageKey: 'testGrid6',
        name: 'propsTest',
        onSort: jest.fn(),
        page: 1,
        refreshGrid: jest.fn(),
        resetGrid: jest.fn(),
        totalItems: 3,
        columns,
        data,
        renderDetailExpansion: jest.fn(),
        renderRowExpansion: jest.fn(),
        showPager: true,
        sort: {
          direction: 'ASC',
          property: 'column2',
        },
      });
      const newColumns = [
        {
          name: 'column1',
          title: 'column1',
          render: rowData => rowData.column1,
          width: 50,
        },
        {
          name: 'column2',
          title: 'column2',
          sortable: true,
          render: rowData => rowData.column2,
          width: 25,
        },
        {
          name: 'column3',
          title: 'column3',
          render: rowData => `${rowData.column3}-displayed`,
          width: 25,
        },
      ];

      it('updateColumns updates the correct state', () => {
        wrapper.setProps({
          columns: newColumns,
        });

        expect(wrapper.state().columns).toEqual(newColumns);
      });
    });

    describe('column reset is triggered', () => {
      const columns = [
        {
          name: 'column1',
          title: 'column1',
          render: rowData => rowData.column1,
          width: 50,
        },
        {
          name: 'column2',
          title: 'column2',
          sortable: true,
          render: rowData => rowData.column2,
          width: 25,
        },
        {
          name: 'column3',
          title: 'column3',
          render: rowData => rowData.column3,
          width: 25,
        },
      ];
      const data = [
        {
          column1: 'stuff1',
          column2: 'more1',
          column3: 'test1',
        },
        {
          column1: 'stuff2',
          column2: 'more2',
          column3: 'test2',
        },
        {
          column1: 'stuff3',
          column2: 'more3',
          column3: 'test3',
        },
      ];
      const resetGridMock = jest.fn();
      const wrapper = fullRender({
        changePage: jest.fn(),
        changePageSize: jest.fn(),
        currentPage: 1,
        emptyText: 'emptyyy',
        expandedDetailIndex: null,
        expandedRowIndices: [],
        isFetching: false,
        loadNextPage: jest.fn(),
        loadPreviousPage: jest.fn(),
        localStorageKey: 'testGrid7',
        name: 'propsTest',
        onSort: jest.fn(),
        page: 1,
        pageLength: 25,
        refresh: jest.fn(),
        resetGrid: resetGridMock,
        totalCount: 3,
        totalPages: 1,
        columns,
        data,
        renderDetailExpansion: jest.fn(),
        renderRowExpansion: jest.fn(),
        showPager: true,
        sort: {
          direction: 'ASC',
          property: 'column2',
        },
      });

      it('resetGrid resets the grid to original props', () => {
        wrapper.find('.bx--grid-panel-settings').simulate('click', {
          stopPropagation: jest.fn(),
        });
        wrapper.find('.bx--grid-panel-settings__reset').simulate('click');

        expect(wrapper.state().columns).toEqual(columns);
        expect(resetGridMock).toBeCalled();
      });
    });

    describe('toggle column visibility is triggered', () => {
      const columns = [
        {
          name: 'column1',
          title: 'column1',
          render: rowData => rowData.column1,
          width: 50,
        },
        {
          name: 'column2',
          title: 'column2',
          sortable: true,
          render: rowData => rowData.column2,
          width: 25,
          hideable: true,
          hidden: false,
        },
        {
          name: 'column3',
          title: 'column3',
          render: rowData => rowData.column3,
          width: 25,
        },
      ];
      const data = [
        {
          column1: 'stuff1',
          column2: 'more1',
          column3: 'test1',
        },
        {
          column1: 'stuff2',
          column2: 'more2',
          column3: 'test2',
        },
        {
          column1: 'stuff3',
          column2: 'more3',
          column3: 'test3',
        },
      ];
      const wrapper = fullRender({
        changePage: jest.fn(),
        changePageSize: jest.fn(),
        currentPage: 1,
        emptyText: 'emptyyy',
        expandedDetailIndex: null,
        expandedRowIndices: [],
        isFetching: false,
        loadNextPage: jest.fn(),
        loadPreviousPage: jest.fn(),
        localStorageKey: 'testGrid8',
        name: 'toggleColumnVisibilityTest',
        onSort: jest.fn(),
        page: 1,
        pageLength: 25,
        refresh: jest.fn(),
        resetGrid: jest.fn(),
        totalCount: 3,
        totalPages: 1,
        columns,
        data,
        renderDetailExpansion: jest.fn(),
        renderRowExpansion: jest.fn(),
        showPager: true,
        sort: {
          direction: 'ASC',
          property: 'column2',
        },
      });

      it('toggleColumnVisibility toggle the appropriate column', () => {
        wrapper.find('.bx--grid-panel-settings').simulate('click', {
          stopPropagation: jest.fn(),
        });
        wrapper.find('.bx--grid-panel-settings__toggle').simulate('click');

        columns[1].hidden = true;
        expect(wrapper.state().columns).toEqual(columns);
      });
    });

    describe('columns are resized', () => {
      const columns = [
        {
          name: 'column1',
          title: 'column1',
          render: rowData => rowData.column1,
          width: 50,
        },
        {
          name: 'column2',
          title: 'column2',
          sortable: true,
          render: rowData => rowData.column2,
          width: 25,
          hideable: true,
          hidden: true,
        },
        {
          name: 'column3',
          title: 'column3',
          render: rowData => rowData.column3,
          width: 25,
        },
      ];
      const data = [
        {
          column1: 'stuff1',
          column2: 'more1',
          column3: 'test1',
        },
        {
          column1: 'stuff2',
          column2: 'more2',
          column3: 'test2',
        },
        {
          column1: 'stuff3',
          column2: 'more3',
          column3: 'test3',
        },
      ];
      const wrapper = fullRender({
        changePage: jest.fn(),
        changePageSize: jest.fn(),
        currentPage: 1,
        emptyText: 'emptyyy',
        expandedDetailIndex: null,
        expandedRowIndices: [],
        isFetching: false,
        loadNextPage: jest.fn(),
        loadPreviousPage: jest.fn(),
        localStorageKey: 'testGrid9',
        name: 'propsTest',
        onSort: jest.fn(),
        page: 1,
        pageLength: 25,
        refresh: jest.fn(),
        resetGrid: jest.fn(),
        totalCount: 3,
        totalPages: 1,
        columns,
        data,
        renderDetailExpansion: jest.fn(),
        renderRowExpansion: jest.fn(),
        showPager: true,
        sort: {
          direction: 'ASC',
          property: 'column2',
        },
      });

      it('updateColumns updates the columns sizes', () => {
        wrapper
          .find('.bx--grid-head-cell__handle')
          .at(0)
          .simulate('dragStart', {
            clientX: 250,
          });
        wrapper.find('.bx--grid-head-cell__handle').at(0).simulate('dragEnd', {
          clientX: 125,
        });

        columns[0].width = 5;
        columns[2].width = 94.99999999999996;
        expect(wrapper.state().columns).toEqual(columns);
      });
    });
  });
});
