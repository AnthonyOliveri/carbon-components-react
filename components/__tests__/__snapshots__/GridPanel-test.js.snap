exports[`GridPanel check if props are rendered correctly standard grid - detail expanded renders correctly 1`] = `
<div
  className="bx--grid-panel">
  <div
    className="bx--grid-panel__pagination">
    <Pagination
      backwardText="Backward"
      className=""
      disabled={false}
      forwardText="Forward"
      itemRangeText={[Function]}
      itemsPerPageText="Items per page"
      onChange={[Function]}
      page={1}
      pageNumberText="Page Number"
      pageRangeText={[Function]}
      pageSizes={
        Array [
          25,
          50,
          75,
          100,
          200,
        ]
      }
      totalItems={3} />
    <button
      className="bx--grid-panel__refresh"
      onClick={[Function]}>
      <Icon
        description="Refresh"
        fillRule="evenodd"
        name="restart--glyph"
        role="img" />
    </button>
    <GridPanelSettings
      items={Array []}
      resetColumns={[Function]}
      toggleColumnVisibility={[Function]} />
  </div>
  <Grid
    columns={
      Array [
        Object {
          "name": "column1",
          "render": [Function],
          "title": "column1",
          "width": 50,
        },
        Object {
          "name": "column2",
          "render": [Function],
          "sortable": true,
          "title": "column2",
          "width": 25,
        },
        Object {
          "name": "column3",
          "render": [Function],
          "title": "column3",
          "width": 25,
        },
      ]
    }
    data={
      Array [
        Object {
          "column1": "stuff1",
          "column2": "more1",
          "column3": "test1",
        },
        Object {
          "column1": "stuff2",
          "column2": "more2",
          "column3": "test2",
          "expander": "expand2",
        },
        Object {
          "column1": "stuff3",
          "column2": "more3",
          "column3": "test3",
          "expander": "expand3",
        },
      ]
    }
    emptyText="emptyyy griddd"
    expandedDetailIndex={2}
    expandedRowIndices={Array []}
    isFetching={false}
    onSort={[Function]}
    renderDetailExpansion={[Function]}
    renderRowExpansion={[Function]}
    sort={
      Object {
        "direction": "ASC",
        "property": "column2",
      }
    }
    updateColumns={[Function]} />
</div>
`;

exports[`GridPanel check if props are rendered correctly standard grid - empty renders correctly 1`] = `
<div
  className="bx--grid-panel">
  <div
    className="bx--grid-panel__pagination">
    <Pagination
      backwardText="Backward"
      className=""
      disabled={false}
      forwardText="Forward"
      itemRangeText={[Function]}
      itemsPerPageText="Items per page"
      onChange={[Function]}
      page={1}
      pageNumberText="Page Number"
      pageRangeText={[Function]}
      pageSizes={
        Array [
          25,
          50,
          75,
          100,
          200,
        ]
      }
      totalItems={3} />
    <button
      className="bx--grid-panel__refresh"
      onClick={[Function]}>
      <Icon
        description="Refresh"
        fillRule="evenodd"
        name="restart--glyph"
        role="img" />
    </button>
    <GridPanelSettings
      items={Array []}
      resetColumns={[Function]}
      toggleColumnVisibility={[Function]} />
  </div>
  <Grid
    columns={
      Array [
        Object {
          "name": "column1",
          "render": [Function],
          "title": "column1",
          "width": 50,
        },
        Object {
          "name": "column2",
          "render": [Function],
          "sortable": true,
          "title": "column2",
          "width": 25,
        },
        Object {
          "name": "column3",
          "render": [Function],
          "title": "column3",
          "width": 25,
        },
      ]
    }
    data={Array []}
    emptyText="emptyyy griddd"
    expandedDetailIndex={null}
    expandedRowIndices={Array []}
    isFetching={false}
    onSort={[Function]}
    renderDetailExpansion={[Function]}
    renderRowExpansion={[Function]}
    sort={
      Object {
        "direction": "ASC",
        "property": "column2",
      }
    }
    updateColumns={[Function]} />
</div>
`;

exports[`GridPanel check if props are rendered correctly standard grid - fetching renders correctly 1`] = `
<div
  className="bx--grid-panel">
  <div
    className="bx--grid-panel__pagination">
    <Pagination
      backwardText="Backward"
      className=""
      disabled={false}
      forwardText="Forward"
      itemRangeText={[Function]}
      itemsPerPageText="Items per page"
      onChange={[Function]}
      page={1}
      pageNumberText="Page Number"
      pageRangeText={[Function]}
      pageSizes={
        Array [
          25,
          50,
          75,
          100,
          200,
        ]
      }
      totalItems={3} />
    <button
      className="bx--grid-panel__refresh"
      onClick={[Function]}>
      <Icon
        description="Refresh"
        fillRule="evenodd"
        name="restart--glyph"
        role="img" />
    </button>
    <GridPanelSettings
      items={Array []}
      resetColumns={[Function]}
      toggleColumnVisibility={[Function]} />
  </div>
  <Grid
    columns={
      Array [
        Object {
          "name": "column1",
          "render": [Function],
          "title": "column1",
          "width": 50,
        },
        Object {
          "name": "column2",
          "render": [Function],
          "sortable": true,
          "title": "column2",
          "width": 25,
        },
        Object {
          "name": "column3",
          "render": [Function],
          "title": "column3",
          "width": 25,
        },
      ]
    }
    data={Array []}
    emptyText="emptyyy"
    expandedDetailIndex={null}
    expandedRowIndices={Array []}
    isFetching={true}
    onSort={[Function]}
    renderDetailExpansion={[Function]}
    renderRowExpansion={[Function]}
    sort={
      Object {
        "direction": "ASC",
        "property": "column2",
      }
    }
    updateColumns={[Function]} />
</div>
`;

exports[`GridPanel check if props are rendered correctly standard grid - hidden pagination toolbar renders correctly 1`] = `
<div
  className="bx--grid-panel">
  <Grid
    columns={
      Array [
        Object {
          "name": "column1",
          "render": [Function],
          "title": "column1",
          "width": 50,
        },
        Object {
          "name": "column2",
          "render": [Function],
          "sortable": true,
          "title": "column2",
          "width": 25,
        },
        Object {
          "name": "column3",
          "render": [Function],
          "title": "column3",
          "width": 25,
        },
      ]
    }
    data={
      Array [
        Object {
          "column1": "stuff1",
          "column2": "more1",
          "column3": "test1",
        },
        Object {
          "column1": "stuff2",
          "column2": "more2",
          "column3": "test2",
        },
        Object {
          "column1": "stuff3",
          "column2": "more3",
          "column3": "test3",
        },
      ]
    }
    emptyText="emptyyy"
    expandedDetailIndex={null}
    expandedRowIndices={Array []}
    isFetching={false}
    onSort={[Function]}
    renderDetailExpansion={[Function]}
    renderRowExpansion={[Function]}
    sort={
      Object {
        "direction": "ASC",
        "property": "column2",
      }
    }
    updateColumns={[Function]} />
</div>
`;

exports[`GridPanel check if props are rendered correctly standard grid - loaded renders correctly 1`] = `
<div
  className="bx--grid-panel">
  <div
    className="bx--grid-panel__pagination">
    <Pagination
      backwardText="Backward"
      className=""
      disabled={false}
      forwardText="Forward"
      itemRangeText={[Function]}
      itemsPerPageText="Items per page"
      onChange={[Function]}
      page={1}
      pageNumberText="Page Number"
      pageRangeText={[Function]}
      pageSizes={
        Array [
          5,
          10,
          15,
        ]
      }
      totalItems={3} />
    <button
      className="bx--grid-panel__refresh"
      onClick={[Function]}>
      <Icon
        description="Refresh"
        fillRule="evenodd"
        name="restart--glyph"
        role="img" />
    </button>
    <GridPanelSettings
      items={Array []}
      resetColumns={[Function]}
      toggleColumnVisibility={[Function]} />
  </div>
  <Grid
    columns={
      Array [
        Object {
          "name": "column1",
          "render": [Function],
          "title": "column1",
          "width": 50,
        },
        Object {
          "name": "column2",
          "render": [Function],
          "sortable": true,
          "title": "column2",
          "width": 25,
        },
        Object {
          "name": "column3",
          "render": [Function],
          "title": "column3",
          "width": 25,
        },
      ]
    }
    data={
      Array [
        Object {
          "column1": "stuff1",
          "column2": "more1",
          "column3": "test1",
        },
        Object {
          "column1": "stuff2",
          "column2": "more2",
          "column3": "test2",
        },
        Object {
          "column1": "stuff3",
          "column2": "more3",
          "column3": "test3",
        },
      ]
    }
    emptyText="emptyyy"
    expandedDetailIndex={null}
    expandedRowIndices={Array []}
    isFetching={false}
    onSort={[Function]}
    renderDetailExpansion={[Function]}
    renderRowExpansion={[Function]}
    sort={
      Object {
        "direction": "ASC",
        "property": "column2",
      }
    }
    updateColumns={[Function]} />
</div>
`;

exports[`GridPanel check if props are rendered correctly standard grid - rows expanded renders correctly 1`] = `
<div
  className="bx--grid-panel">
  <div
    className="bx--grid-panel__pagination">
    <Pagination
      backwardText="Backward"
      className=""
      disabled={false}
      forwardText="Forward"
      itemRangeText={[Function]}
      itemsPerPageText="Items per page"
      onChange={[Function]}
      page={1}
      pageNumberText="Page Number"
      pageRangeText={[Function]}
      pageSizes={
        Array [
          25,
          50,
          75,
          100,
          200,
        ]
      }
      totalItems={3} />
    <button
      className="bx--grid-panel__refresh"
      onClick={[Function]}>
      <Icon
        description="Refresh"
        fillRule="evenodd"
        name="restart--glyph"
        role="img" />
    </button>
    <GridPanelSettings
      items={Array []}
      resetColumns={[Function]}
      toggleColumnVisibility={[Function]} />
  </div>
  <Grid
    columns={
      Array [
        Object {
          "name": "column1",
          "render": [Function],
          "title": "column1",
          "width": 50,
        },
        Object {
          "name": "column2",
          "render": [Function],
          "sortable": true,
          "title": "column2",
          "width": 25,
        },
        Object {
          "name": "column3",
          "render": [Function],
          "title": "column3",
          "width": 25,
        },
      ]
    }
    data={
      Array [
        Object {
          "column1": "stuff1",
          "column2": "more1",
          "column3": "test1",
        },
        Object {
          "column1": "stuff2",
          "column2": "more2",
          "column3": "test2",
          "expander": "expand2",
        },
        Object {
          "column1": "stuff3",
          "column2": "more3",
          "column3": "test3",
          "expander": "expand3",
        },
      ]
    }
    emptyText="emptyyy griddd"
    expandedDetailIndex={null}
    expandedRowIndices={
      Array [
        1,
        2,
      ]
    }
    isFetching={false}
    onSort={[Function]}
    renderDetailExpansion={[Function]}
    renderRowExpansion={[Function]}
    sort={
      Object {
        "direction": "ASC",
        "property": "column2",
      }
    }
    updateColumns={[Function]} />
</div>
`;
